bug: app crash when clicked on No Thanks in flexible update

**Plugin version:**
1.3.0



**Platform(s):**
Android - Observed on Android 10 and 11


**Current behavior:**
when the user taps on No Thanks , when flexible update popup is shown , the app crashes , i dont think it is happening all the time but we could reproduce it majority of the times.

Here's the crash report

java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:5579)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:5620)
  at android.app.servertransaction.ActivityResultItem.execute (ActivityResultItem.java:51)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2325)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8633)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
Caused by: java.lang.NullPointerException: 
  at com.google.android.play.core.internal.aw.a (Unknown Source:7)
  at com.google.android.play.core.listener.b.b (Unknown Source:14)
  at com.google.android.play.core.appupdate.e.unregisterListener (Unknown Source:3)
  at dev.robingenz.capacitor.appupdate.AppUpdatePlugin.handleOnActivityResult (AppUpdatePlugin.java:169)
  at com.getcapacitor.Bridge.onActivityResult (Bridge.java:1010)
  at com.getcapacitor.BridgeActivity.onActivityResult (BridgeActivity.java:215)
  at android.app.Activity.dispatchActivityResult (Activity.java:8550)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:5572)



**Expected behavior:**
The popup should just dismiss.


**Steps to reproduce:**
just tap on No Thanks in the flexible app update and you receive a crash.



**Related code:**
<!-- Code which helps to reproduce or better understand the issue. -->

```
insert short code snippets here
```

**Other information:**
<!-- List any other information that is relevant to your issue. Device information, stack traces, screenshots, related issues, etc. -->



**Capacitor doctor:**

💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 3.4.0
  @capacitor/core: 3.4.0
  @capacitor/android: 3.4.0
  @capacitor/ios: 3.4.0

Installed Dependencies:

  @capacitor/cli: 3.2.5
  @capacitor/core: 3.2.5
  @capacitor/android: 3.2.4
  @capacitor/ios: 2.5.0

--- a/android/src/main/java/dev/robingenz/capacitor/appupdate/AppUpdatePlugin.java
+++ b/android/src/main/java/dev/robingenz/capacitor/appupdate/AppUpdatePlugin.java
@@ -154,10 +154,7 @@ public class AppUpdatePlugin extends Plugin {
 
     @PluginMethod
     public void completeFlexibleUpdate(PluginCall call) {
-        if (this.listener != null) {
-            this.appUpdateManager.unregisterListener(this.listener);
-            this.listener = null;
-        }
+        this.unregisterListener();
         this.appUpdateManager.completeUpdate();
         call.resolve();
     }
@@ -166,8 +163,7 @@ public class AppUpdatePlugin extends Plugin {
     protected void handleOnActivityResult(int requestCode, int resultCode, Intent data) {
         super.handleOnActivityResult(requestCode, resultCode, data);
         if (resultCode != RESULT_OK && requestCode == REQUEST_FLEXIBLE_UPDATE) {
-            this.appUpdateManager.unregisterListener(this.listener);
-            this.listener = null;
+            this.unregisterListener();
         }
         this.appUpdateInfo = null;
         if (savedPluginCall == null) {
@@ -207,5 +203,13 @@ public class AppUpdatePlugin extends Plugin {
             return false;
         }
         return true;
+    }
+
+    private void unregisterListener() {
+        if (this.listener == null || this.appUpdateManager == null) {
+            return;
+        }
+        this.appUpdateManager.unregisterListener(this.listener);
+        this.listener = null;
     }
 }

